library/std/src/panic.rs

```diff
@@ -316,16 +316,18 @@ where
 ///
 /// # Notes
 ///
-/// Note that this function **might not catch all panics** in Rust. A panic in
-/// Rust is not always implemented via unwinding, but can be implemented by
-/// aborting the process as well. This function *only* catches unwinding panics,
-/// not those that abort the process.
-///
-/// Note that if a custom panic hook has been set, it will be invoked before
-/// the panic is caught, before unwinding.
-///
-/// Also note that unwinding into Rust code with a foreign exception (e.g.
-/// an exception thrown from C++ code) is undefined behavior.
+/// This function **might not catch all panics** in Rust. A panic in Rust is not
+/// always implemented via unwinding, but can be implemented by aborting the
```

Comment:
Would something like "in Rust" -> "from Rust" or "started by Rust" be more accurate?

Reply:
I'm not sure - it's really talking about the behavior of the panic handler here, and the fact that `panic!` and `panic_any` can abort the process rather than unwinding. It seems to me that it could simply say:

> This function **cannot** catch panics when `panic=abort`, or with a manually written panic handler that aborts the process.

...but it could be that there's some reason why it doesn't allude to `panic=abort` here (I admit I don't really know what the conventions are for the standard library documentation and whether they can refer to compiler config).

Reply:
I've gone ahead and replaced this paragraph with the sentence I proposed above, explicitly referring to `panic=abort`.

Reply:
Discussion moved to https://github.com/rust-lang/rust/pull/128321#discussion_r1720949174